Source/WebCore/loader/DocumentWriter.cpp

```diff
@@ -241,7 +241,7 @@ bool DocumentWriter::begin(const URL& urlReference, bool dispatch, Document* own
     if (existingDocument && existingDocument->contentSecurityPolicy() && document->contentSecurityPolicy())
         document->checkedContentSecurityPolicy()->setInsecureNavigationRequestsToUpgrade(existingDocument->checkedContentSecurityPolicy()->takeNavigationRequestsToUpgrade());
 
-    frameLoader->didBeginDocument(dispatch);
+    frameLoader->didBeginDocument(dispatch, triggeringAction ? triggeringAction->navigationAPIType() : std::nullopt);
```

Comment:
I don't like this part. Can FrameLoader::didBeginDocument request this information? The less files we change the better.